File: src/main/java/ac/kr/deu/connect/luck/food_truck/entity/FoodTruckMenu.java
package ac.kr.deu.connect.luck.food_truck.entity;

import ac.kr.deu.connect.luck.common.BaseEntity;
import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class FoodTruckMenu extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    @Column(columnDefinition = "LONGTEXT")
    private String imageUrl;
    private int price;

    @JsonBackReference
    @ManyToOne
    private FoodTruck foodTruck;
}

File: src/main/java/ac/kr/deu/connect/luck/foodtruck/entity/FoodTruckMenu.kt
package ac.kr.deu.connect.luck.foodtruck.entity

import jakarta.persistence.Embeddable
import java.math.BigDecimal

@Embeddable
data class FoodTruckMenu(
    val name: String,
    val description: String?,
    val image: String?,
    val price: BigDecimal?,
)
